### ISSUE TITLE
Linking issue with Elixir on Alpine ARM

### ISSUE THREAD

**Issue by tombruijn** [appsignal/appsignal-elixir#795]
We got a report that our extension fails to install, more specifically link, on Alpine Linux using an ARM container.

It works for the Ruby gem, so this issue is Elixir specific.

## To do

- [x] Find a way to fix the linking error
- [ ] Find a workaround if not possible, by downgrading either our package or Alpine Linux
  - It's probably possible to avoid this by not using Alpine Linux on ARM, but instead on a non-ARM host, or not using Alpine Linux.

## Linking error

```
Build error was encountered while running `make` (exit code: 2):

/usr/lib/gcc/aarch64-alpine-linux-musl/11.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: /app/_build/dev/lib/appsignal/priv/libappsignal.a(fp_mode.o): in function `__fe_getround':
/cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.45/./lib/builtins/fp_mode.c:20: multiple definition of `__fe_getround'; /app/_build/dev/lib/appsignal/priv/libappsignal.a(fp_mode.o):/cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.45/./lib/builtins/fp_mode.c:20: first defined here
/usr/lib/gcc/aarch64-alpine-linux-musl/11.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: /app/_build/dev/lib/appsignal/priv/libappsignal.a(fp_mode.o): in function `__fe_raise_inexact':
/cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.45/./lib/builtins/fp_mode.c:24: multiple definition of `__fe_raise_inexact'; /app/_build/dev/lib/appsignal/priv/libappsignal.a(fp_mode.o):/cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.45/./lib/builtins/fp_mode.c:24: first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:43: all] Error 1
```

Duplicate symbols:

- `__fe_getround`
- `__fe_raise_inexact`

## Files to reproduce

Dockerfile
```Dockerfile
# Dockerfile
FROM --platform=linux/arm64/v8 elixir:1.14-alpine

RUN apk add --update alpine-sdk coreutils

RUN mix local.hex --force && \
    mix local.rebar --force
```

Test script, run in this repository:
```sh
#!/bin/bash
# ./run

set -eu

chmod +x inner_run
rm -rf _build deps
make clean

IMAGE_NAME=elixir_alpine_test
docker build --platform linux/arm64/v8 -t elixir_alpine_test:latest .
docker run -v "$(pwd)":"/app" -w "/app" --rm --platform linux/arm64/v8 -it elixir_alpine_test:latest ./inner_run
```
(I put platform everywhere just to make sure it uses the right one always.)

Commands to run:
```sh
#!/bin/ash
# ./inner_run
mix deps.get
mix appsignal.diagnose --send-report
```

## Resources

- [Possible location of the definition of these symbols](https://github.com/rust-lang/compiler-builtins/blob/83b2c07a6bfa394fa5979ca12b60536f402a792c/build.rs#L436-L437)

## Conversations

- [Intercom conversation](https://app.intercom.com/a/inbox/yzor8gyw/inbox/admin/4356044/conversation/16410700147085?view=List) (private link)

**Comment by tombruijn** [comment]
The following seems to help: `LDFLAGS += -Wl,-z muldefs`, but that's declaring it okay for any duplicates to be ignored. I don't know if that can have an adverse impact or not.

**Comment by tombruijn** [comment]
Released the fix for this in version 2.2.19.
